chainer/optimizer.py

```diff
@@ -27,153 +30,313 @@ def exponential_decay_noise(xp, shape, dtype, hook, opt):
     return xp.random.normal(0, std, shape).astype(dtype)
 
 
-class Optimizer(object):
-    """Base class of all numerical optimizers.
+class Hyperparameter(object):
 
-    This class provides basic features for all optimization methods. It
-    optimizes parameters of a *target link*. The target link is registered via
-    the :meth:`setup` method, and then the :meth:`update` method updates its
-    parameters based on a given loss function.
+    """Set of hyperparameters of an optimizer.
 
-    Each optimizer implementation must be defined as a child class of
-    Optimizer. It must override :meth:`update` method. An optimizer can use
-    *internal states* each of which is tied to one of the parameters. State is
-    a dictionary of serializable values (typically arrays of size same as
-    the corresponding parameters). In order to use state dictionaries, the
-    optimizer must override :meth:`init_state` method (or its CPU/GPU versions,
-    :meth:`init_state_cpu` and :meth:`init_state_gpu`).
+    This is a utility class to provide a set of hyperparameters for update
+    rules and an optimizer. It can be used as a usual object.
```

Comment:
Thank you. Then, I think it is more kind to explain the expected usage like "You can add hyperparameters as attributes as a usual object".